modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/blueprint/parameter/SXPParameterDataCreator.java

```diff
@@ -104,26 +93,8 @@ public SXPParameterData create(
 		return new SXPParameterData(keywords, sxpParameters);
 	}
 
-	@Override
-	public List<SXPParameterContributorDefinition>
-		getSXPParameterContributorDefinitions(long companyId, Locale locale) {
-
-		if (ArrayUtil.isEmpty(_sxpParameterContributors)) {
-			return Collections.emptyList();
-		}
-
-		List<SXPParameterContributorDefinition>
-			sxpParameterContributorDefinitions = new ArrayList<>();
-
-		for (SXPParameterContributor sxpParameterContributor :
-				_sxpParameterContributors) {
-
-			sxpParameterContributorDefinitions.addAll(
-				sxpParameterContributor.getSXPParameterContributorDefinitions(
-					companyId, locale));
-		}
-
-		return sxpParameterContributorDefinitions;
+	public SXPParameterContributor[] getSxpParameterContributors() {
+		return _sxpParameterContributors;
```

Comment:
Hey @RenanUmV, this method seems off into the `SXPParameterDataCreator`. Let's try this:

1.  extract this logic to a `SXPParameterContributorProvider` class and call it into the `SXPParameterContributorDefinitionProviderImpl`. 
2. Then rename the `SXPParameterDataCreator` to `SXPParameterDataCreatorUtil` and make this a simple Util class.
3. Pass the `SXPParameterContributorProvider` as a parameter to the `SXPParameterDataCreatorUtil.create` method